### PR TITLE
bug 1801397: Make existing pdb work

### PR DESCRIPTION
**Pull request by deads2k**
builds on https://github.com/openshift/cluster-etcd-operator/pull/157 with a goal of making the existing PDBs functional by copying the content "first" in the listing

### CODE REVIEW

File: bindata/etcd/pod.yaml [hunk 1]
```diff
@@ -5,9 +5,37 @@ metadata:
...
+      terminationMessagePolicy: FallbackToLogsOnError
+      command:
+        - /bin/sh
+        - -c
+        - |
+          #!/bin/sh
+          set -euo pipefail
+
+          cp /etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.crt /etc/kubernetes/etcd-backup-dir/system:etcd-peer-NODE_NAME.crt
+          cp /etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.key /etc/kubernetes/etcd-backup-dir/system:etcd-peer-NODE_NAME.key
```
Comment (hexfusion) [inline bindata/etcd/pod.yaml]: FTR the CA needed for quorum-guard is still on disk in `static-pod-resources/etcd-member/ca.crt` if anyone was concerned we don't see a CA being copied.

**Comment by deads2k** [comment]
message: 'Failed to create pod sandbox: rpc error: code = Unknown desc = error reserving
    pod name k8s_apiserver-2vvft_openshift-apiserver_8017545a-12fd-473d-8106-061bc789273f_0
    for id cd5c6debe69a2f1fd934a6f3c64231dea390d515b36c0aa730666edb3cea4cf3: name
    is reserved'
  metadata:

/retest

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/162/pull-ci-openshift-cluster-etcd-operator-master-e2e-gcp/565

File: pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go [hunk 2]
```diff
@@ -159,7 +168,34 @@ func (c *BootstrapTeardownController) removeBootstrap() error {
...
 
+	// check to see if bootstrapping is complete
+	bootstrapFinished := false
+	bootstrapFinishedConfigMap, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("bootstrap", metav1.GetOptions{})
+	if err != nil {
+		klog.V(2).Infof("non-fatal error getting configmap: %v", err)
+	} else {
+		if bootstrapFinishedConfigMap.Data["status"] == "complete" {
+			bootstrapFinished = true
+		}
```
Comment (ericavonb) [inline pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go]: golang cleanup nit:
```suggestion
		bootstrapFinished = bootstrapFinishedConfigMap.Data["status"] == "complete"
```

File: pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go [hunk 3]
```diff
@@ -159,7 +168,34 @@ func (c *BootstrapTeardownController) removeBootstrap() error {
...
+			klog.V(2).Infof("non-fatal error getting bootstrap-finished event: %v", err)
+		} else {
+			bootstrapFinished = true
+		}
+	}
+
+	if !bootstrapFinished {
+		c.eventRecorder.Event("DelayingBootstrapTeardown", "cluster-bootstrap is not yet finished")
+		return nil
+	}
```
Comment (ericavonb) [inline pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go]:
```suggestion
		c.eventRecorder.Event("DelayingBootstrapTeardown", "cluster-bootstrap is not yet finished")
		return nil
	}
}
```

Comment (hexfusion) [reply pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go]: thanks! we will circle back around for cleanup. I appreciate the notes.

**Pull request merged**